### ISSUE TITLE
object-feature-part

### ISSUE THREAD

**Issue by VladimirAlexiev** [linked-art/linked.art#84]
http://linked.art/model/object/physical/#features
- "features have no objectively-definable border, such as the recto and verso of the canvas".
But there is an objectively-definable border, namely the edge of the painting.
I think all features have objectively-definable borders: this is where the feature ends.
So I'd restate it something like this: "unlike parts, features are not physically separable, eg the recto and verso of the canvas".
- more importantly, I don't see the reason for nesting a part (ManMadeObject) under the feature.
  - The whole reason for using Feature is that it's not physically separable from the object, so why add a MMO under it? The examples given "Photographic Part" and "Back Part" suggest these are separate parts of the object, which defeats the purpose of using Feature
  - just like MMO, Features can also depict subjects and carry images or text. It says "The artwork or content that is part of the front or back" but content is a conceptual object not physical.
  - "This is to provide consistency with the more common situation when the features are not modeled at all": I think that object-part vs object-feature emphasizes the symmetry of the two situations (physically separable part vs non-separable feature), and I don't see why object-feature-part is "more consistent"

**Comment by azaroth42** [comment]
This is taken not quite verbatim from the scope notes of E26:
>  ... there are no natural borders that separate them [features] completely in an objective way from the carrier objects.

So thanks for the suggestion, but I think it's accurate as is.

The nesting under the feature is to ensure that artworks that are not modeled with recto/verso features are still MMOs, as you quote. If symmetry with the the part relationship is more important, then we should just not use features and treat recto/verso as parts.

**Comment by VladimirAlexiev** [comment]
Then please quote a bit more verbatim. It talks of impossibility to separate features from carrier objects, whereas your example suggests you can't separate features from each other, and gives an unfortunate example (recto/verso) where you can.

> ensure that artworks that are not modeled with recto/verso features are still MMOs

How does the nesting artwork-feature-part ensure that?
My point is that "Photographic (Front) Part" and "Back Part" are completely artificial (non-existent) Physical objects (unless you split the photo in two with a very thin razor). Can you define what these MMOs **are**?

**Comment by azaroth42** [comment]
I think that the recto side is impossible to separate -- as soon as you removed the "recto", it would become a different object as there would be a new side one would consider recto.  From two sides on one object there would be four sides on two objects.  The rationale for feature is, thus, that you can't remove the "recto" and still have it retain any notion of "recto-ness". 

The MMOs that are part of the features are artificial. They're there so that artworks are MMOs, because that's how they're typically thought of in systems.  If there's a panel with a different painting on each side, from different artists, we want those paintings to behave like paintings that are on a panel with nothing of interest at all on the other side.  There are two options -- require recto/verso for everything (and what that would mean for more complex 3d shapes, I don't know), or include the artificial "artwork" object.  We went with the latter.

**Comment by workergnome** [comment]
I suppose the other option (not that I'm suggesting this) is to open the Pandora's Box where Artworks are no longer MMOs, but Conceptual Objects.  But that would have massive ripple effects across the whole model.

**Comment by VladimirAlexiev** [comment]
@azaroth42 
1. Artworks are MMO that 
2. (may bear features that) 
3. depict things, or 
  - carry images or other information objects that 
    - refer to things

There's no second level of MMO anywhere there. 
The verso does not have part an Artwork (MMO). You don't need such artificial MMO (and it does not exist): the verso itself carries that image.

**Comment by VladimirAlexiev** [comment]
Another example: "Nelson died on this spot of the upper deck of RMS Victory". 
If you want to model the particular **board** of the deck (maybe because you found a bullet hole there), you better go ship-hasPart-deck-hasPart-board.
If you don't need to model the parts of the ship, better go ship-hasFeature-deck.

**Comment by azaroth42** [comment]
That would mean that there are two distinct patterns with the optional inclusion of features in the middle, to solve a 1% level problem. I'd rather ignore features all together than make the model more complex for the 99% case.

I'll work up a recto/verso as part version.

**Comment by VladimirAlexiev** [comment]
You already have a distinct pattern (MMO vs MMO-feature-MMO). I'm just arguing

**Comment by ewg118** [comment]
Obverse and Reverse are considered Features of the Man-Made Object (presuming a coin, and not another non-man-made sort of currency like a shell)

**Comment by VladimirAlexiev** [comment]
@ewg118 We know that. But both Feature and MMO are subclass of E18 Physical Thing. Question is would you be happy with using just this class for simplicity.
Plus added P2_has_type pointing to AAT concept (coin, reverse, obverse).

**Comment by VladimirAlexiev** [comment]
@azaroth42 A downside of a single class: how do you find all the artworks and only the artworks?
This is not a trivial question even if you use several different classes, since parts are also MMO but they're not artworks.

By ownership? You'd own an artwork, but probably not its parts, and definitely not its features.

**Comment by ewg118** [comment]
I think it would be okay for the sake of simplicity to use P2_has_type to distinguish between the coin, obverse, and reverse MMOs.

The artwork discussion has come up on the slack list. I'm not sure what the final answer was.

**Comment by azaroth42** [comment]
Looking at the class hierarchy, E22 is the only class that allows properties with a domain/range of E19 (Physical Object), E24 (Physical Man-Made Thing) and its parent E71 (Man-Made Thing)

E71 is the domain of:  P102 has title, and P103 was intended for.
E24 is the domain of P62 depicts, P65 shows visual item.
E19 is the domain of: P54 has current permanent location, P55 has current location, P56 bears feature and P57 has number of parts, and the range of P25 moved.

Man Made Feature is a E18 (parent of E19) and E24, thus missing the E19 predicates (E22 doesn't have any of its own).

E19 is also the "all aggregates of objects" class that replaces E78.
 
The simplest route, to me, is to have all parts of MMOs also be MMOs, including ones that CRM would want us to call Features. Then artworks can be `aat:300133025`, recto/verso have their own classifications (), and we don't need to worry about people understanding when to use Feature versus MMO versus Physical Object.  We have a very simple rule:

Any part of any man made object is also a man made object. Any collection of objects is an E19.  
If the MMO is an artwork, it MUST be classified_as aat:300133025. The different parts of the object should have different classificiations, such as frame/support, recto/verso or obverse/reverse, as appropriate to the object's main classification (painting vs manuscript vs coin).

Collections thus do not depict things (which is correct, their members do) nor have titles (which is fine, as we don't use P102 and Title anyway). To Vlad's original point, if you take a laser and cut the coin in half, you still have two MMOs.

**Comment by azaroth42** [comment]
azaroth42 marks this issue as being discussed in [slack](https://linked-art.slackarchive.io/-/search-7ee0ef6333d544dfa8c6241fa9eb73b4/page-1)

**Comment by VladimirAlexiev** [comment]
Good domain/range analysis. But think also about [Mixins](https://en.wikipedia.org/wiki/Mixin): these are classes that "appear when you need them".
- eg consider Legal Object. Since we we had full RDFS reasoning for British Museum, we had like 20M legal objects given 2.5M museum objects ("Lawyers of the world, rejoice!"), and BM didn't even have any Rights objects!! The mixin philosophy would have it that if and only if an object has a Right, it is considered a Legal object: much more satisfying.

How about using E18 Physical Thing? Many museums carry non-man-made things that are not easy to identify as such. So in many cases MMO would be a misnomer class.

E18 doesn't have Production (a non-man-made object is not Produced by an agent). 
- But the mixin philosophy would have it that if you tack Production on E18, then (eg by RDFS domain reasoning) it also becomes E22 MMO.

E18 also doesn't have depict: CRM presumes that only man can depict or represent things, but nature cannot. It's not obvious this should be so, see this delightful story:
- http://nerdist.com/these-crabs-look-like-samurai-warriors-and-it-might-be-our-fault/
"Over the centuries, Japanese fishermen have released any Heikeposis crabs whose carapaces (in their eyes) resembled Samurai battle masks. Some postulate that in doing so, the fishermen actually influenced the direction of natural selection."

**Comment by VladimirAlexiev** [comment]
Apart from domain/range, the other important line of reasoning is "as we use fewer classes and lose selectivity, what can go wrong". Eg
- how do you find all the artworks and only the artworks?

**Comment by azaroth42** [comment]
Discussed at f2f as a possible option

**Comment by azaroth42** [comment]
Here, my argument for keeping the profile as it is, and not using Features at all.

A common case is a manuscript codex with hinges, jewels, clasps or other objects that are part of the covers. Just do a google image search for "manuscript clasp".

TLDR:   Only features can be part of features, but there are clear use cases where this is not the case if we use feature for recto/verso, such as clasps, hinges, decorations, curtains, foldouts, paste-ins, etc. etc.

Consider, for example: https://art.thewalters.org/detail/26557 

* The codex itself is clearly a single E22 Man Made Object.
* The front cover is objectively definable, so an E22 which is part of (P46) the codex. [See note below]
* The front of the front cover is not separable or objectively definable, so would be a Feature which is born by (P56) the front cover MMO.  The same as the recto and verso of the pages within it.
* The clasp is part of the front of the front cover. And now we hit the ontological paradox, that a feature would now contain an object, yet P56 bears feature is a short cut for a Place based definition, which allows features to only be related to a single physical object. In this case the front of the front cover is born by both the later brass clasp and the older ["alum-tawed skin"](https://www.ligatus.org.uk/lob/concept/1197). This would be the same as foldouts, curtains, paste-downs, and so forth for the codex pages. 

Note - You might disagree that the Front Cover is objectively separable from the spine or the Back Cover, and thus a feature. Which doesn't affect the argument, as it just introduces a Feature for the Front cover, which then is partitioned into two features, front of front cover and back of front cover.

You could say that there is an aggregate object that is the single object for the front that bears the feature, and that aggregate is composed of the skin part and the brass clasp part ... but then you have an objectively defined object, which as it is also produced by human activity, would really be an E22 ... bringing us back to where we are now anyway.

So, despite the presence of Features in the ontology, I argue that it is more consistent and less complicated to always use ManMadeObject without introducing invalid inferential data (and we don't consider inference as a necessary quality anyway). Further, the use of features does introduce paradoxes in common use cases.

**Comment by natuk** [comment]
Forgive me but I do not see why it needs to be modeled like that. The book (E22 Man-Made Object) has different parts (P46 composed of): a) the cover of the book (E22 Man-Made Object) and b) the clasp (E22 Man-Made Object). The cover (E22 Man-Made Object) bears feature (P56) the front (left) cover. The clasp (E22) is located (P55 has current location) the front foredge (E53 Place). The clasp is not part of the front cover.

**Comment by azaroth42** [comment]
Thanks Thanasis, no need for forgiveness! :)

Part of the brass clasp is visible on the front cover... but I grant you that it's not entirely part of the cover. Imagine that it's instead a precious stone set into the cover ... does the same apply?  Or for a necklace of coins -- they're in the necklace place, rather than being parts?

In the use of a relative E53 Place for attachments, what's the connection between the E22 Book and the E53? P59 has section, per the long-cut of bears feature?

For the Place pattern, it seems to lack any sort of intent or permanence.  The clasp (or stone) is intended to be "on" the front cover (and leading edge, and back cover) permanently. It's attached, not just resting there. I could put my coffee cup on the front cover, and it would be in the same location with the same predicate.  P54 has current permanent location is maybe overly specific in the scope note and example, but could be applied to distinguish from P55?

I could definitely see Place as a model for physical containment, (the E22 letter is the in the E53 place that is a section of [the E25 inside of] the E22 box).

**Comment by azaroth42** [comment]
My understanding, in a diagram:
<img width="639" alt="feature-place_vs_mmo" src="https://user-images.githubusercontent.com/871868/55017140-5c6b8e80-4fad-11e9-8c9e-d62eb7a01551.png">

**Comment by natuk** [comment]
I like the illustration! "this" is a "[centre](http://w3id.org/lob/concept/2848)" 
OK, I was thinking that the permanence aspect of the clasp being attached to something is given by the statement that it is part of the book. I appreciate P55 was designed with other cases in mind although the scope note does not exclude permanent positions. Reading the thread again, where is the ontological paradox, though? Why can't we simply use P46 between the front cover (feature) and the clasp? Am I wrong in thinking that P46 is inherited by E26 Physical Feature?
Having said that I still think that the clasp is not part of the cover, but the situation may apply to other cases.

**Comment by VladimirAlexiev** [comment]
The front of the front cover (the recto of the cover) doesn't have a third dimension, so it's not a physical thing, and you can't attach anything to it. A clasp or jewel cannot be set into something that has no physical depth.

This is enough to state the `<center>` is affixed to the front cover:
```ttl
<book> a MMO; has part <cover>.
<cover> a MMO; has part <center>.
<center> a MMO.
```
(sorry, I'm too lazy to look up the class & prop numbers).

If you also want to state the `<center>` is located at the recto of the cover, you'd need to add:
```ttl 
<cover> bears feature <front of cover as feature>.
<front of cover as feature> section <front of cover as place>.
<front of cover as place> location of <center>
```

**Comment by natuk** [comment]
I broadly agree but features can have three dimensions, e.g. embossed tooling on leather covers. Also feature inherits P46 so it can have its own parts which do not have to be features, no? E.g. a doorway and its wooden frame.

**Comment by azaroth42** [comment]
I propose that in order to change to using feature+place, there would need to be a clear use case for something that cannot be expressed with the simpler model, that data actually exists for. Otherwise we're just documenting the universe and falling into the same trap as the full CRM model.

(See [design principles](https://linked.art/model/profile/))

**Comment by ewg118** [comment]
Insofar as feature+place is concerned, we have a lot of examples on coins where we want/need to designate the location of a feature on the surface of the obverse or reverse of a coin. There are commonly accepted terms within all branches of numismatics, such as "exergue" (below the head/ground line of an iconographic motif), "left field," "right field," etc. Other "places" might be arbitrarily defined, based on terminologies developed by one scholar for one category of coinage (e.g., just the coinage of Alexander the Great). Some of these are "below the throne [of Zeus]", "underneath of the wing [of Nike, presumably]". Location of a symbol or monogram on the obverse/reverse of a coin may have semantic meaning. One symbol located in one place may have a different meaning than the same symbol located in another place.

I don't know that you could say a coin in an MMO that has a side that is an MMO that has multiple symbols that are also MMOs.

**Comment by workergnome** [comment]
What about the very common occurrence in museum data where we have multiple dimensions for a given object?

Take http://browse.americanartcollaborative.org/object/ima/5063.html, for instance.  There is a sheet dimension and an image dimension.  Given the MMO pattern, we would describe the sheet as a MMO, that has a part (MMO) that is the image.  

This strikes me as a very common occurrence in data--even more so that objects with actual, physical parts, and to me strikes me as a very strong argument for features.

**Comment by illip** [comment]
If my understanding is good, the issue here is to decide if whether or not we want to allow one or two ways for partitioning. 

e22 -> p46 -> e22: This ceramic teapot lid is part of this ceramic teapot. 
I think we all agree that we need this path.

e22 -> p46/p56 -> e25: This ceramic teapot handle is part of this ceramic teapot.  
From my point of view this path is more complicated to manage because:

1) I don't know how I will differenciate, in my data, a "feature handle" like the ceramic one and a "MMO handle" which is screwed to a metal teapot. So even if I have the opportunities to use e25, I would probably model everything to the first path.

2) I believe that, in a lot of cases, the choice between e22 and e25 is debatable. In my previous example, someone could argue that my screwed handle is a e25. I would probably prefer parsing the data using e55 based on a good partitioning vocabulary than hoping to get the right info with e25. 

Also, I think there is no property attached to e22 and e25 so we are not losing meaningful properties if we decide to limit LA partitioning to e22. However, I totally understand that by doing this we are not totally following the scope note of e22 but since we can expect that e25 path will probably be misused and that RDF shapes would probably be unuseful here, I'm not sure if it's a good idea to open this door.

That said, I totally understand the reason why e25 is a well suited class for coins but again, I'm not sure we are losing a lot of semantic by using e22. For the image dimensions, I think we would have e36 -> p43 -> e54. However if we really want to express the dimension of a blue painting spot on this canvas or the provenance of this paint, we get in what I call the "concrete material issue". For our first target model here at CHIN (that we didn't implement yet), we decided to use e25 only in concrete material cases. In other words, we use e22 -> p46 -> e25 -> p45 -> e57 to be able to describe the conceptual material (e57) and the concrete material (e25). For us, it seems possible to always identify "features" in the materiality context. That said, the benefit is to be more aligned on the scope notes because we could have also used e22 -> p2 -> e55 (Concrete Material).

Also from my understanding, I can still located a part by using e22 -> p59 -> e53.

So for me, the distinction between e22 and e25 would be very complicated to implement and would result in a messier dataset.

**Comment by azaroth42** [comment]
:100: to Philippe! I think the cost of using Features outweighs the gains, especially as it's often impossible to tell from the data, which pattern to use.

I propose that we close the issue and leave it as is, with partitioning only via E22 Man Made Object. The goal of the profile is to reduce complexity and thereby increase adoption and implementation. The barrier for introducing complexity such as this needs to be that there are *use cases that cannot otherwise be fulfilled, that institutions have actual data to support*. We haven't seen that, but can re-open it if there is such. Given the close relationship in the ontology between Feature and Man Made Object, I would be very surprised if there is actual functionality that cannot be represented with a MMO combined with a p2_has_type.

**Comment by zeroexp** [comment]
we have different types of properties in our system using lio (lightweight image ontology) which first relates the objects in the image to the image itself. http://www.imagesnippets.com/ArtSpeak/help/properties.html. 

Where you have been delineating each item as a man made object  - we have distinctions such as: depicts and 'shows' that indicate the level of importance in describing the object(s) in the image. We can say something like: This Image Depicts a 'teapot' which hasMaterials ceramic and this image depicts a 'lid' hasMaterials 'ceramic'  and you can also add the triple to each of these that each of these items is a man-made object from the crm if desired. So the triple would be: this image depicts teapot which is <crm:E22> I do personally question why it is necessary to say that practically every object is man-made when the data itself make the man-made distinction as a superclass in many areas.  (i.e. in general, a teapot is going to be 'man-made' and would fall into a class of things that are man-made).

**Comment by azaroth42** [comment]
Agreed to close, no change.
